src/js/jwt/insights/offline.js

```diff
@@ -50,18 +50,19 @@ export function getOfflineToken(realm, clientId) {
     return Promise.reject('not available');
   }
 
-  const tokenURL = `${insightsUrl(DEFAULT_ROUTES)}/realms/${realm}/protocol/openid-connect/token`;
+  const ssoUrl = await insightsUrl(DEFAULT_ROUTES);
+
+  const tokenURL = `${ssoUrl}/realms/${realm}/protocol/openid-connect/token`;
   const params = parseHashString(postbackUrl);
 
-  return axios({
-    method: 'post',
-    url: tokenURL,
-    data: getPostDataString(getPostDataObject(postbackUrl, clientId, params.code)),
-    config: { headers: { 'Content-Type': 'application/x-www-form-urlencoded' } },
-  }).then((response) => {
-    priv.response = response;
-    return response;
-  });
+  return axios
+    .post(tokenURL, getPostDataObject(postbackUrl, clientId, params.code), {
```

Comment:
@karelhala Looks like axios.post() with object doesn't do the right thing for form-urlencoded, it sends JSON.  
I'm seeing errors from SSO for `insights.chrome.auth.getOfflineToken()` in [qaprodauth](https://qaprodauth.console.redhat.com/openshift/token/show), [qaprodauth beta](https://qaprodauth.console.redhat.com/beta/openshift/token/show), and [prod-beta](https://console.redhat.com/beta/openshift/token/show);
[prod-stable](https://console.redhat.com/openshift/token/show) works.  

old working request has payload looking like `code=...&grant_type=authorization_code&client_id=cloud-services&redirect_uri=https%3A%2F%[2Fconsole.redhat.com](http://2Fconsole.redhat.com)%2Fopenshift%2Ftoken%2Fshow%3Fnoauth%3D2402500adeacc30eb5c5a8a5e2e0ec1f`.
new failing request has JSON payload `{"code":"...","grant_type":"authorization_code","client_id":"cloud-services","redirect_uri":"https%3A%2F%[2Fconsole.redhat.com](http://2Fconsole.redhat.com)%2Fbeta%2Fopenshift%2Ftoken%2Fshow%3Fnoauth%3D2402500adeacc30eb5c5a8a5e2e0ec1f"}`
This causes server to respond 400 Bad Request, body `{"error":"invalid_request","error_description":"Missing form parameter: grant_type"}` plus omit Access-Control-Allow-Origin header so what browser sees is CORS error.

Per https://github.com/axios/axios#using-applicationx-www-form-urlencoded-format, passing an object is not supposed to work, see there for alternatives (or bring back `getPostDataString()`).

Reply:
@cben  This change was not pushed to prod-stable yet. @karelhala can you please check this and fix the issue?

Reply:
Sure thing!

Reply:
@cben thank you for the investigation! I've overlooked the form url encoded format and just blindly removed the encoding part. This PR brings back the encoding function and sends correct data https://github.com/RedHatInsights/insights-chrome/pull/1759

Reply:
Thanks.
- [x] https://qaprodauth.console.redhat.com/beta/openshift/token/show now works again (and we haven't pushed any app changes there, so confirms the insights-chrome fix :+1:)
- [ ] https://qaprodauth.console.redhat.com/openshift/token/show still sends JSON.
- [ ] https://console.redhat.com/beta/openshift/token/show still sends JSON.  I assume those just weren't deployed yet.

Reply:
We will promote the changes to other envs soon.